### ISSUE TITLE
Stub only package not found when installed in editable mode

### ISSUE THREAD

**Issue by Holt59** [microsoft/pyright#5290]
**Describe the bug**

I have a stub only package, which I can disclose here, but with a structure similar to this:

```
mypackage-stubs/
    __init__.pyi
    py.typed
    sub/
        __init__.pyi
        aa.pyi
    ...
pyproject.toml
```

When I install it using `pip install .`, the stubs are properly found by pyright, but when installing in editable mode with `pip install -e .`, the stubs are reported as missing.

**Expected behavior**

I'd like the stubs to be available when installing in editable mode.

**VS Code extension or command-line**

I'm using the VS Code extension.

**Comment by erictraut** [comment]
Please refer to [this documentation](https://microsoft.github.io/pyright/#/import-resolution?id=editable-installs) for editable installs. This should take care of your problem. If it doesn't, let me know, and I'll re-open this issue.

**Comment by Holt59** [comment]
I looked at the documentation and tried a 

```bash
# editable_mode=strict does not work either
pip install -e . --config-settings editable_mode=compat
```

A `.pth` file was created under `site-packages` with a name like `__editable__.mypackage-version.pth` but pyright still reports missing stubs.

**Comment by mrrezaie** [comment]
`python -m pip install -e . --config-settings editable_mode=compat`

This could handle that. It would be great if you add this to the documentation.

Thanks.
